clicommand/artifact_upload.go

```diff
@@ -147,12 +153,13 @@ var ArtifactUploadCommand = cli.Command{
 
 		// Setup the uploader
 		uploader := agent.NewArtifactUploader(l, client, agent.ArtifactUploaderConfig{
-			JobID:          cfg.Job,
-			Paths:          cfg.UploadPaths,
-			Destination:    cfg.Destination,
-			ContentType:    cfg.ContentType,
-			DebugHTTP:      cfg.DebugHTTP,
-			FollowSymlinks: cfg.FollowSymlinks,
+			JobID:            cfg.Job,
+			Paths:            cfg.UploadPaths,
+			Destination:      cfg.Destination,
+			ContentType:      cfg.ContentType,
+			DebugHTTP:        cfg.DebugHTTP,
+			FollowSymlinks:   cfg.FollowSymlinks,
+			NoUploadSymlinks: !cfg.UploadSymlinks,
```

Comment:
We need to invert the bool as the existing behaviour is equivalent to setting `NoUploadSymlinks` to false, and when creating a new `agent.ArtifactUploaderConfig` struct, the field will be `false` by default.